### DISCUSSION TITLE
Is federation spec version 1 still supported?

### DISCUSSION THREAD

**Discussion by sefi18** [Netflix/dgs-framework#1233] [Q&A] [answered]
Hello everybody,

I noticed that DGS started to support graphql federation version 2, which is really great!
We are currently evaluating a federation gateway that supports only version 1 and therefore we have to be backwards compatible.

Does DGS still support federation v1? Are there any explicit settings I have to make?

Example:

If I extend the query type in an v1 compatible manner it ends up in an error at runtime:

```
extend type Query {
...
}
```
(see also v1 spec: https://www.apollographql.com/docs/federation/v1/quickstart#example-subgraphs)

I get this error at runtime:
`Caused by: com.netflix.graphql.dgs.exceptions.InvalidDgsConfigurationException: Parent type Query not found, but it was referenced on com.netflix.graphql.dgs.internal.DgsSchemaProvider in @DgsData annotation for field permissions`

Downgrading the federation-graphql-java-support maven library to version 0.9.0 does not make any difference.

Would be happy if someone could help...
Thanks in advance :-)

**Comment by srinivasankavitha** [comment, marked as answer]
v1 is very much supported and actively in use internally. We actually
haven't moved to federation 2.0 functionality, it just has bare minimum
support and is backwards compatible.

Your error looks unrelated to federation. In order to extend a "Query" you
need a parent "Query" to be defined in your schema to begin with. That is
what you are missing. Hope that helps.

**Comment by sefi18** [comment]
@srinivasankavitha Thanks for clarifying this!

**Comment by dastin-sandura-a1** [comment]
I have the same error when starting test execution in Intelij IDEA. I do have the Query as the parent type. This is my graphql schema:

type Query{
    users(ids: [Int!]!): [User!]!
}
type User {
    id: Int!
}

How to fix it in this situation? What is the cause?
